torch/csrc/autograd/function.h

```diff
@@ -387,10 +464,18 @@ struct TORCH_API Node : std::enable_shared_from_this<Node> {
   /// Calls `apply()`, but instruments it with tracing machinery.
   variable_list traced_apply(variable_list inputs);
 
-  // Since `Node`s are neither copyable nor moveable, we can have const
-  // fields.
+  // Sequence number used to correlate backward nodes with forward ops in the
+  // profiler and provide determinisim in the engine.
   const uint64_t sequence_nr_;
 
+  // See NOTE [ Topological Number ]
+  uint64_t topological_nr_ = 0;
+
+  // Tracks whether this node has been added as the next_edge of another node
+  // via set_next_edge(s), which always calls topological_nr() of all its children
+  // See NOTE [ Topological Number ] for why we need this.
+  mutable bool has_parent_ = false;
```

Comment:
why mutable?

Reply:
I have the getter for topological_nr as const